### PR TITLE
ci: add dependency-checker command

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
Add a dependency checker tool that can provide a summary of dependencies which need to be updated. This includes platform support.

Right now this can just be run locally, but the intent is to have it run on a schedule in CI and create issues about dependencies needing updates.

Credit goes to @kkujawa-sumo and https://github.com/kkujawa-sumo/gardener-helper, of which this is a copy with some minor bug fixes.

### Checklist

- [x] Changelog updated or skip changelog label added
- [x] Documentation updated

### CODE REVIEW

**Comment by swiatekm-sumo** [comment]
Current output:

```bash
$ make check-dependencies
Gardener helper
#####################################################################

#### kOps ####
kOps latest versions from https://github.com/kubernetes/kops/releases
['1.24', '1.25', '1.26', '1.27', '1.28']
Currently supported kOps versions for Sumologic Kubernetes Collection Helm Chart
['1.23', '1.24', '1.25', '1.26', '1.27']
Expected supported kOps versions for Sumologic Kubernetes Collection Helm Chart
['1.24', '1.25', '1.26', '1.27', '1.28']

Please add support to following kOps versions:
['1.28']
Please remove support to following kOps versions:
['1.23']

#### EKS ####
EKS officially supported versions
['1.24', '1.25', '1.26', '1.27', '1.28']
Currently supported EKS versions for Sumologic Kubernetes Collection Helm Chart
['1.23', '1.24', '1.25', '1.26', '1.27']


Please add support to following EKS versions:
['1.28']
Please remove support to following EKS versions:
['1.23']

#### GKE ####
GKE officially supported versions
['1.25', '1.26', '1.27']
Currently supported GKE versions for Sumologic Kubernetes Collection Helm Chart
['1.24', '1.25', '1.26', '1.27']


Please remove support to following GKE versions:
['1.24']

#### AKS ####
AKS officially supported versions
['1.24', '1.25', '1.26', '1.27*', '1.28']
Currently supported AKS versions for Sumologic Kubernetes Collection Helm Chart
['1.25', '1.26', '1.27']


Please add support to following AKS versions:
['1.24', '1.27*', '1.28']
Please remove support to following AKS versions:
['1.27']

#### OpenShift ####
OpenShift officially supported versions
['4.11', '4.12', '4.13', '4.14']
Currently supported OpenShift versions for Sumologic Kubernetes Collection Helm Chart
['4.10', '4.11', '4.12', '4.13']


Please add support to following OpenShift versions:
['4.14']
Please remove support to following OpenShift versions:
['4.10']

#### Subcharts in Sumologic Kubernetes Collection Helm Chart ####

Please check newer version of KUBE-PROMETHEUS-STACK subchart, version: 52.1.0, created: 2023-10-26T21:48:10.210926072+00:00
Currently used KUBE-PROMETHEUS-STACK subchart, version: 40.5.0, created: 2022-10-07T09:23:39.496033848+00:00

Please check newer version of FALCO subchart, version: 3.8.4, created: 2023-10-27T12:09:17.669775109+00:00
Currently used FALCO subchart, version: 3.3.0, created: 2023-06-29T16:34:18.10367103+00:00

Please check newer version of METRICS-SERVER subchart, version: 6.5.5, created: 2023-10-11T08:52:05.848005567+00:00
Currently used METRICS-SERVER subchart, version: 6.4.3, created: 2023-06-22T15:37:29.691326752+00:00

Please check newer version of TAILING-SIDECAR-OPERATOR subchart, version: 0.9.0, created: 2023-07-20T13:03:15.444380115+00:00
Currently used TAILING-SIDECAR-OPERATOR subchart, version: 0.8.0, created: 2023-07-05T13:20:26.224786501+00:00

Please check newer version of OPENTELEMETRY-OPERATOR subchart, version: 0.41.0, created: 2023-10-19T16:11:13.404996572+00:00
Currently used OPENTELEMETRY-OPERATOR subchart, version: 0.35.0, created: 2023-08-10T15:49:17.866060304+00:00
```

**Review by mat-rumian — APPROVED** [review]
Really cool stuff 🚀

**Comment by mat-rumian** [comment]
@swiatekm-sumo not sure if AKS looks fine, I mean version `1.27*` vs `1.27`

**Review by rnishtala-sumo — APPROVED** [review]

**Comment by swiatekm-sumo** [comment]
@mat-rumian fixed the AKS reporting, it now reads:

```md
#### AKS ####
AKS officially supported versions
['1.25', '1.26', '1.27', '1.28']
Currently supported AKS versions for Sumologic Kubernetes Collection Helm Chart
['1.25', '1.26', '1.27']

```

File: Makefile [hunk 1]
```diff
@@ -53,6 +53,10 @@ markdown-links-lint:
 check-configuration-keys:
 	./ci/check_configuration_keys.py --values deploy/helm/sumologic/values.yaml --readme deploy/helm/sumologic/README.md
 
+.PHONY: check-dependencies
+check-dependencies:
```
Comment (kkujawa-sumo) [inline Makefile:57]: on Vagrant this target doesn't work:
```
$ make check-dependencies
python ./ci/check_dependencies/main.py
make: python: Command not found
make: *** [Makefile:58: check-dependencies] Error 127
```

Comment (kkujawa-sumo) [reply Makefile:57]: Ok, I see readme below :)

Comment (swiatekm-sumo) [reply Makefile:57]: I think you have an outdated machine. Do a `vagrant destroy` and `vagrant up` and it should work.

**Review by kkujawa-sumo — APPROVED** [review]
Thank you for adding this here! ❤️ ❤️ ❤️

**Pull request merged**